SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -244,18 +272,84 @@ TArray<UClass*> GetAllSupportedClasses()
 	return Classes.Array();
 }
 
-bool SpatialGDKGenerateSchema()
+void DeleteGeneratedSchemaFiles()
 {
-	ClassPathToSchema.Empty();
+	const FString SchemaOutputPath = GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder();
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	if (PlatformFile.DirectoryExists(*SchemaOutputPath))
+	{
+		if (!PlatformFile.DeleteDirectoryRecursively(*SchemaOutputPath))
+		{
+			UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Could not clean the generated schema directory '%s'! Please make sure the directory and the files inside are writeable."), *SchemaOutputPath);
+		}
+	}
+	PlatformFile.CreateDirectory(*SchemaOutputPath);
+}
 
-	SchemaGeneratedClasses = GetAllSupportedClasses();
+void InitClassPathToSchemaMap()
+{
+	TSoftObjectPtr<USchemaDatabase> SchemaDatabasePtr(FSoftObjectPath(TEXT("/Game/Spatial/SchemaDatabase.SchemaDatabase")));
+	SchemaDatabasePtr.LoadSynchronous();
+	const USchemaDatabase* const SchemaDatabase = SchemaDatabasePtr.Get();
+
+	if (SchemaDatabase)
+	{
+		ClassPathToSchema = SchemaDatabase->ClassPathToSchema;
+		NextAvailableComponentId = SchemaDatabase->NextAvailableComponentId;
+	}
+	else
+	{
+		// As a safety precaution, if the SchemaDatabase.uasset doesn't exist then make sure the schema generated folder is cleared as well. 
+		DeleteGeneratedSchemaFiles();
+	}
+}
+#pragma optimize("", off)
+void PreProcessSchemaMap()
+{
+	TArray<FString> EntriesToRemove;
+	for (const auto& EntryIn : ClassPathToSchema)
+	{
+		UE_LOG(LogSpatialGDKSchemaGenerator, Warning, TEXT("Checking if following file exists: %s"), *(EntryIn.Key));
+
+		const FSoftObjectPath ItemToReference(EntryIn.Key);
+
+		// first check if the object is already loaded into memory
+		UObject* const ResolvedObject = ItemToReference.ResolveObject();
+		UObject* const LoadedObject   = ResolvedObject ? nullptr : ItemToReference.TryLoad();
+
+		// only store classes that weren't currently loaded into memory
+		if (LoadedObject && !AdditionalSchemaGeneratedClasses.Find(Cast<UClass>(LoadedObject)))
+		{
+			// don't allow the Garbage Collector to delete these objects
+			LoadedObject->AddToRoot();
```

Comment:
Worth adding a note that these objects will *never* get cleaned up, even if moving between maps.